connector/src/main/java/org/geysermc/connector/network/translators/java/entity/JavaEntitySetPassengersTranslator.java

```diff
@@ -87,7 +88,13 @@ public void translate(ServerEntitySetPassengersPacket packet, GeyserSession sess
             if (entity.getEntityType() == EntityType.BOAT) {
                 passenger.getMetadata().put(EntityData.RIDER_ROTATION_LOCKED, (byte) 1);
                 passenger.getMetadata().put(EntityData.RIDER_MAX_ROTATION, 90f);
-                passenger.getMetadata().put(EntityData.RIDER_MIN_ROTATION, !passengers.isEmpty() ? -90f : 0f);
+                // Can be removed once 1.16.200 to 1.16.201 support is dropped
+                if (session.getUpstream().getSession().getPacketCodec().getProtocolVersion() >= Bedrock_v428.V428_CODEC.getProtocolVersion()) {
+                    passenger.getMetadata().put(EntityData.RIDER_MIN_ROTATION, 1f);
+                    passenger.getMetadata().put(EntityData.RIDER_ROTATION_OFFSET, -90f);
+                } else {
+                    passenger.getMetadata().put(EntityData.RIDER_MIN_ROTATION, -90f);
```

Comment:
Does the `passengers.isEmpty` check not apply at all anymore?

Reply:
`passengers` will never be empty because of line 84.

Reply:
Ah, nice catch.